src/devices/machine/sis630_host.h

```diff
@@ -0,0 +1,99 @@
+// license: BSD-3-Clause
+// copyright-holders: Angelo Salese
+
+#ifndef MAME_MACHINE_SIS630_HOST_H
+#define MAME_MACHINE_SIS630_HOST_H
+
+#pragma once
+
+#include "pci.h"
+#include "sis630_gui.h"
+
+class sis630_host_device : public pci_host_device
+{
+public:
+	sis630_host_device(
+		const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock,
+		const char *cpu_tag, int ram_size
```

Comment:
You should template it on the CPU tag type so it can take object finders rather than only literal tags.